### PR TITLE
[MNOE-483] Several Fixes

### PR DESCRIPTION
**Pull request by x4d3**
- Add SubTenant Config Flag
- Clear Client Relation when updating user to admin
- Add more information in SubTenant Clients

### CODE REVIEW

File: api/app/assets/javascripts/mno_enterprise/config.js.coffee.erb [hunk 1]
```diff
@@ -18,6 +18,7 @@ angular.module('mnoEnterprise.configuration', [])
   .constant('ADMIN_PANEL_CONFIG', <%= Hash(Settings.admin_panel).to_json %>)
   .constant('PAYMENT_CONFIG', <%= Hash(Settings.payment).to_json %>)
   .constant('ORGANIZATION_MANAGEMENT', <%= Hash(Settings.organization_management).to_json %>)
+  .constant('SUB_TENANT_CONFIG', <%= Hash(Settings.sub_tenant).to_json %>)
```
Comment (ouranos) [inline api/app/assets/javascripts/mno_enterprise/config.js.coffee.erb]: Could we nest that under `ADMIN_PANEL_CONFIG`?
That's easier to manage that'll be easier for v2 migration.

File: core/lib/generators/mno_enterprise/install/templates/config/settings.yml [hunk 2]
```diff
@@ -57,6 +57,9 @@ user_management:
 devise:
   registration:
     disabled: false
+# SubTenant Feature
+sub_tenant:
```
Comment (ouranos) [inline core/lib/generators/mno_enterprise/install/templates/config/settings.yml]: Please nest under `admin_config`

**Comment by x4d3** [comment]
@ouranos I applied your change request.

File: api/app/controllers/mno_enterprise/jpi/v1/admin/users_controller.rb [hunk 3]
```diff
@@ -87,7 +87,14 @@ def user_update_params
         attrs << :admin_role
         attrs << :mnoe_sub_tenant_id
       end
-      params.require(:user).permit(attrs)
+      user_param = params.require(:user)
+      updated_params = user_param.permit(attrs)
+      updated_params[:client_ids] ||= [] if user_param.has_key?(:client_ids)
```
Comment (ouranos) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/users_controller.rb]: Is this line needed?

**Comment by ouranos** [comment]
@x4d3 could you have a look at my previous comment?

File: api/app/controllers/mno_enterprise/jpi/v1/admin/users_controller.rb (hunk 3 above)
Comment (x4d3) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/users_controller.rb]: I was getting nil instead of an empty array for client_ids

**Pull request merged**